### DISCUSSION TITLE
A faster way to check if a record exists in a Postgres table that matches a condition

### DISCUSSION THREAD

**Discussion by JohnnyRacer** [sqlalchemy/sqlalchemy#10404] [Usage Questions]
Hello, I want to query my table to check if a certain record exists that matches a certain condition without returning the query result. From what I understand, SQL `SELECT` has O(log(n)) time complexity. Is there a way to check if a record exists with a faster alternative like O(n) or O(1) ideally?

**Comment by CaselIT** [comment]
Hi,

You can look into indexes and exists, running `select exists (your select here)` will return only a boolean
